servers/src/main/java/tachyon/worker/block/BlockMetadataManager.java

```diff
@@ -127,6 +127,12 @@ public void addTempBlockMeta(TempBlockMeta tempBlockMeta)
    */
   public void commitTempBlockMeta(TempBlockMeta tempBlockMeta)
       throws WorkerOutOfSpaceException, BlockAlreadyExistsException, BlockDoesNotExistException {
+    long blockId = tempBlockMeta.getBlockId();
+    if (hasBlockMeta(blockId)) {
```

Comment:
this looks like a bug, however, this may be affecting speculative execution. @calvinjia , is the original code intended?